app/mesh/1.4.x/features/kds-auth.md

```diff
@@ -161,9 +161,43 @@ Verify the remote control plane is connected with authentication by looking at t
 2021-02-24T14:30:38.596+0100	INFO	kds.auth	Remote CP successfully authenticated using Control Plane Token	{"tokenSerialNumber": 1, "zone": "cluster-2"}
 ```
 
-## Rotate tokens
+## Revoke token
 
-If a control plane token or signing key is compromised, you must rotate all tokens.
+Kong Mesh does not keep the list of issued tokens. Whenever a single token is compromised, we can add it to revocation list so it's no longer valid.
+
+Every token has its own ID which is available in payload under `jti` key. You can extract ID from token using jwt.io or [`jwt-cli`](https://www.npmjs.com/package/jwt-cli) tool. Here is example of `jti`
+```
+0e120ec9-6b42-495d-9758-07b59fe86fb9
+```
+
+Specify list of revoked IDs separated by `,` and store it as `GlobalSecret` named `control-plane-token-revocations`
+
+{% navtabs %}
+{% navtab Kubernetes %}
+```sh
+REVOCATIONS=$(echo '0e120ec9-6b42-495d-9758-07b59fe86fb9' | base64) && echo "apiVersion: v1
+kind: Secret
+metadata:
+  name: control-plane-token-revocations
+  namespace: kuma-system 
+data:
+  value: $REVOCATIONS
+type: system.kuma.io/global-secret" | kubectl apply -f -
+```
+{% endnavtab %}
+{% navtab Universal %}
+```sh
+echo "
+type: GlobalSecret
+name: control-plane-token-revocations
+data: {{ revocations }}" | kumactl apply --var revocations=$(echo '0e120ec9-6b42-495d-9758-07b59fe86fb9' | base64) -f -
+```
+{% endnavtab %}
+{% endnavtabs %}
+
+## Rotate signing key
+
+If a signing key is compromised, you must rotate it and all the tokens.
```

Comment:
If a signing key is comprimised, would that automatically invalidate all tokens once rotated, or do you also have to add all of those tokens to the revocation list?

Reply:
Once you remove the compromised signing key, all tokens signed with this signing key are no longer valid so you don't have to manually add them to the revocation list.

See `Remove the old signing key` - `All new connections to the global control plane now require tokens signed with the new signing key`